src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  readonly grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function gerAverageGrades(student: Student): number {
+  return (
+    student.grades.reduce((acc, grade) => acc + grade) / student.grades.length
+  );
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        return sortedStudents.sort(
+          (st1, st2) => st1[sortBy].localeCompare(st2[sortBy]),
+        );
+      }
+
+      return sortedStudents.sort(
+        (st1, st2) => st1[sortBy].localeCompare(st2[sortBy]),
+      );
```

Comment:
Can be shorter here. Also do this for all of the `case`s
```suggestion
  const sortOrder = order === 'asc' ? 1 : -1;
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return sortedStudents.sort(
        (st1, st2) => sortOrder * st1[sortBy].localeCompare(st2[sortBy]),
      );
```